### ISSUE TITLE
Errors in loading state_dict for ARTrackV2

### ISSUE THREAD

**Issue by G1tQuality** [MIV-XJTU/ARTrack#92]
作者您好，请问[ARTrackV2-B-256](https://drive.google.com/file/d/1tGaY5jQxZOTzJDWXgOgoHtBwc5l4NLQ2/view?usp=drive_link) 链接中的文件名的确是OSTrack_ep0040吗？
我下载权重放到pretrained_model目录下，并修改config.yaml文件以及https://github.com/MIV-XJTU/ARTrack/blob/d082c85361686d65ae26b35f685890686d5cc5d1/lib/test/parameter/artrackv2.py#L24 处checkpoint的路径，在运行`python tracking/test.py artrackv2 artrackv2_256_full --dataset lasot --threads 4 --num_gpus 1`后出现如下报错：
`Evaluating    1 trackers on     1 sequences
Tracker: artrackv2 artrackv2_256_full None ,  Sequence: Ves-9
test config:  {'MODEL': {'PRETRAIN_FILE': '/home/gaoletian/ARTrack/pretrained_model/OSTrack_ep0040.pth.tar', 'PRETRAIN_PTH': '', 'EXTRA_MERGER': False, 'RETURN_INTER': False, 'RETURN_STAGES': [2, 5, 8, 11], 'BACKBONE': {'TYPE': 'vit_base_patch16_224', 'STRIDE': 16, 'MID_PE': False, 'SEP_SEG': False, 'CAT_MODE': 'direct', 'MERGE_LAYER': 0, 'ADD_CLS_TOKEN': False, 'CLS_TOKEN_USE_MODE': 'ignore', 'CE_LOC': [], 'CE_KEEP_RATIO': [], 'CE_TEMPLATE_RANGE': 'ALL'}, 'BINS': 400, 'RANGE': 2, 'EXTENSION': 3, 'ENCODER_LAYER': 3, 'NUM_HEADS': 12, 'MLP_RATIO': 4, 'QKV_BIAS': True, 'DROP_RATE': 0.1, 'ATTN_DROP': 0.0, 'DROP_PATH': 0.0, 'DECODER_LAYER': 6, 'HEAD': {'TYPE': 'PIX', 'NUM_CHANNELS': 768}}, 'TRAIN': {'LR': 8e-05, 'WEIGHT_DECAY': 0.0001, 'EPOCH': 100, 'LR_DROP_EPOCH': 40, 'BATCH_SIZE': 16, 'NUM_WORKER': 6, 'OPTIMIZER': 'ADAMW', 'BACKBONE_MULTIPLIER': 0.1, 'GIOU_WEIGHT': 2.0, 'L1_WEIGHT': 0.0, 'SCORE_WEIGHT': 0.0, 'FREEZE_LAYERS': [0], 'PRINT_INTERVAL': 10, 'VAL_EPOCH_INTERVAL': 20, 'GRAD_CLIP_NORM': 0.1, 'AMP': False, 'CE_START_EPOCH': 20, 'CE_WARM_EPOCH': 80, 'DROP_PATH_RATE': 0.1, 'SCHEDULER': {'TYPE': 'step', 'DECAY_RATE': 0.1}}, 'DATA': {'SAMPLER_MODE': 'causal', 'MEAN': [0.485, 0.456, 0.406], 'STD': [0.229, 0.224, 0.225], 'MAX_SAMPLE_INTERVAL': 200, 'TRAIN': {'DATASETS_NAME': ['LASOT'], 'DATASETS_RATIO': [1], 'SAMPLE_PER_EPOCH': 2000}, 'VAL': {'DATASETS_NAME': ['GOT10K_votval'], 'DATASETS_RATIO': [1], 'SAMPLE_PER_EPOCH': 200}, 'SEARCH': {'SIZE': 256, 'FACTOR': 4.0, 'CENTER_JITTER': 3, 'SCALE_JITTER': 0.25, 'NUMBER': 1}, 'TEMPLATE': {'NUMBER': 2, 'SIZE': 128, 'FACTOR': 2.0, 'CENTER_JITTER': 0, 'SCALE_JITTER': 0}}, 'TEST': {'TEMPLATE_FACTOR': 2.0, 'TEMPLATE_SIZE': 128, 'SEARCH_FACTOR': 4.0, 'SEARCH_SIZE': 256, 'EPOCH': 100}}
Error(s) in loading state_dict for ARTrackV2:
        Unexpected key(s) in state_dict: "cross_2_decoder.mask_token", "cross_2_decoder.decoder_pos_embed", "cross_2_decoder.decoder_embed.weight", "cross_2_decoder.decoder_embed.bias", "cross_2_decoder.decoder_blocks.0.norm1.weight", "cross_2_decoder.decoder_blocks.0.norm1.bias", "cross_2_decoder.decoder_blocks.0.attn.qkv.weight", "cross_2_decoder.decoder_blocks.0.attn.qkv.bias", "cross_2_decoder.decoder_blocks.0.attn.proj.weight", "cross_2_decoder.decoder_blocks.0.attn.proj.bias", "cross_2_decoder.decoder_blocks.0.norm2.weight", "cross_2_decoder.decoder_blocks.0.norm2.bias", "cross_2_decoder.decoder_blocks.0.mlp.fc1.weight", "cross_2_decoder.decoder_blocks.0.mlp.fc1.bias", "cross_2_decoder.decoder_blocks.0.mlp.fc2.weight", "cross_2_decoder.decoder_blocks.0.mlp.fc2.bias", "cross_2_decoder.decoder_blocks.1.norm1.weight", "cross_2_decoder.decoder_blocks.1.norm1.bias", "cross_2_decoder.decoder_blocks.1.attn.qkv.weight", "cross_2_decoder.decoder_blocks.1.attn.qkv.bias", "cross_2_decoder.decoder_blocks.1.attn.proj.weight", "cross_2_decoder.decoder_blocks.1.attn.proj.bias", "cross_2_decoder.decoder_blocks.1.norm2.weight", "cross_2_decoder.decoder_blocks.1.norm2.bias", "cross_2_decoder.decoder_blocks.1.mlp.fc1.weight", "cross_2_decoder.decoder_blocks.1.mlp.fc1.bias", "cross_2_decoder.decoder_blocks.1.mlp.fc2.weight", "cross_2_decoder.decoder_blocks.1.mlp.fc2.bias", "cross_2_decoder.decoder_blocks.2.norm1.weight", "cross_2_decoder.decoder_blocks.2.norm1.bias", "cross_2_decoder.decoder_blocks.2.attn.qkv.weight", "cross_2_decoder.decoder_blocks.2.attn.qkv.bias", "cross_2_decoder.decoder_blocks.2.attn.proj.weight", "cross_2_decoder.decoder_blocks.2.attn.proj.bias", "cross_2_decoder.decoder_blocks.2.norm2.weight", "cross_2_decoder.decoder_blocks.2.norm2.bias", "cross_2_decoder.decoder_blocks.2.mlp.fc1.weight", "cross_2_decoder.decoder_blocks.2.mlp.fc1.bias", "cross_2_decoder.decoder_blocks.2.mlp.fc2.weight", "cross_2_decoder.decoder_blocks.2.mlp.fc2.bias", "cross_2_decoder.decoder_blocks.3.norm1.weight", "cross_2_decoder.decoder_blocks.3.norm1.bias", "cross_2_decoder.decoder_blocks.3.attn.qkv.weight", "cross_2_decoder.decoder_blocks.3.attn.qkv.bias", "cross_2_decoder.decoder_blocks.3.attn.proj.weight", "cross_2_decoder.decoder_blocks.3.attn.proj.bias", "cross_2_decoder.decoder_blocks.3.norm2.weight", "cross_2_decoder.decoder_blocks.3.norm2.bias", "cross_2_decoder.decoder_blocks.3.mlp.fc1.weight", "cross_2_decoder.decoder_blocks.3.mlp.fc1.bias", "cross_2_decoder.decoder_blocks.3.mlp.fc2.weight", "cross_2_decoder.decoder_blocks.3.mlp.fc2.bias", "cross_2_decoder.decoder_blocks.4.norm1.weight", "cross_2_decoder.decoder_blocks.4.norm1.bias", "cross_2_decoder.decoder_blocks.4.attn.qkv.weight", "cross_2_decoder.decoder_blocks.4.attn.qkv.bias", "cross_2_decoder.decoder_blocks.4.attn.proj.weight", "cross_2_decoder.decoder_blocks.4.attn.proj.bias", "cross_2_decoder.decoder_blocks.4.norm2.weight", "cross_2_decoder.decoder_blocks.4.norm2.bias", "cross_2_decoder.decoder_blocks.4.mlp.fc1.weight", "cross_2_decoder.decoder_blocks.4.mlp.fc1.bias", "cross_2_decoder.decoder_blocks.4.mlp.fc2.weight", "cross_2_decoder.decoder_blocks.4.mlp.fc2.bias", "cross_2_decoder.decoder_blocks.5.norm1.weight", "cross_2_decoder.decoder_blocks.5.norm1.bias", "cross_2_decoder.decoder_blocks.5.attn.qkv.weight", "cross_2_decoder.decoder_blocks.5.attn.qkv.bias", "cross_2_decoder.decoder_blocks.5.attn.proj.weight", "cross_2_decoder.decoder_blocks.5.attn.proj.bias", "cross_2_decoder.decoder_blocks.5.norm2.weight", "cross_2_decoder.decoder_blocks.5.norm2.bias", "cross_2_decoder.decoder_blocks.5.mlp.fc1.weight", "cross_2_decoder.decoder_blocks.5.mlp.fc1.bias", "cross_2_decoder.decoder_blocks.5.mlp.fc2.weight", "cross_2_decoder.decoder_blocks.5.mlp.fc2.bias", "cross_2_decoder.decoder_blocks.6.norm1.weight", "cross_2_decoder.decoder_blocks.6.norm1.bias", "cross_2_decoder.decoder_blocks.6.attn.qkv.weight", "cross_2_decoder.decoder_blocks.6.attn.qkv.bias", "cross_2_decoder.decoder_blocks.6.attn.proj.weight", "cross_2_decoder.decoder_blocks.6.attn.proj.bias", "cross_2_decoder.decoder_blocks.6.norm2.weight", "cross_2_decoder.decoder_blocks.6.norm2.bias", "cross_2_decoder.decoder_blocks.6.mlp.fc1.weight", "cross_2_decoder.decoder_blocks.6.mlp.fc1.bias", "cross_2_decoder.decoder_blocks.6.mlp.fc2.weight", "cross_2_decoder.decoder_blocks.6.mlp.fc2.bias", "cross_2_decoder.decoder_blocks.7.norm1.weight", "cross_2_decoder.decoder_blocks.7.norm1.bias", "cross_2_decoder.decoder_blocks.7.attn.qkv.weight", "cross_2_decoder.decoder_blocks.7.attn.qkv.bias", "cross_2_decoder.decoder_blocks.7.attn.proj.weight", "cross_2_decoder.decoder_blocks.7.attn.proj.bias", "cross_2_decoder.decoder_blocks.7.norm2.weight", "cross_2_decoder.decoder_blocks.7.norm2.bias", "cross_2_decoder.decoder_blocks.7.mlp.fc1.weight", "cross_2_decoder.decoder_blocks.7.mlp.fc1.bias", "cross_2_decoder.decoder_blocks.7.mlp.fc2.weight", "cross_2_decoder.decoder_blocks.7.mlp.fc2.bias", "cross_2_decoder.decoder_norm.weight", "cross_2_decoder.decoder_norm.bias", "cross_2_decoder.decoder_pred.weight", "cross_2_decoder.decoder_pred.bias", "backbone.pos_embed_z", "backbone.prev_position_embeddings.weight". 
Done`

请问作者问题可能出在哪里，十分感谢您能回答我的问题！

**Comment by AlexDotHam** [comment]
你comfig可能没用对，我们的权重对应的是artrackv2 seq 的那一套config，因为你这里开的是一阶段训练的模型，因此是没有appearance的decoder的，希望能帮助到你
